mantidproject/mantid
# Add extra load file tooltip on Linux aimed at IDAaaS users

In the new year (2024), the way that users access the archive on ISAaaS will be changing slightly, so they'll need to mount the archive themselves using a new shortcut in the Applications menu. This extra addition to the tooltip will hopefully help them out a bit.

### Description of work
- When searching for a file path, instead of just returning a string (or an empty string for a fail), a ``Result<std::string>`` is returned. This new type wraps the result (if there is one) with any error messages.
- In ``ISISDataArchive``, if it manages to get the path by querying ``data.isis.rl.ac.uk``, but then can't actually get to that file on your machine, then you'll see the new message about IDAaaS.

Fixes #36566. 




### To test:
- New tooltip should only appear on Linux with the archive option set to ``on`` or ``ISIS``.
- Try and load a file that's in the archive, but with the archive not mounted, the new message should appear.







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
